federation-js/src/composition/validate/sdl/uniqueTypeNamesWithFields.ts

```diff
@@ -121,9 +121,38 @@ export function UniqueTypeNamesWithFields(
           return false;
         });
 
+      const inputValuesDiff = Object.entries(inputValues);
+
+      const typesHaveSameInputValuesShape =
+        inputValuesDiff.length === 0 ||
+        inputValuesDiff.every(([name, types]) => {
+          if (types.length === 2) {
+            possibleErrors.push(
+              errorWithCode(
+                'VALUE_TYPE_INPUT_VALUE_MISMATCH',
```

Comment:
Can we get a test case that exercises this code path and demonstrates what will trigger this error?

Reply:
done! I wish we could access the field name to improve the error message, but alas I wasn't able to :)

Reply:
I see. I think we can find a way to pass that info down through the diff but I'm really happy with where we've landed here. This feels ready to land to me and I don't want to block getting this released. If you don't mind opening an issue that references this PR comment that would be wonderful!

Reply:
done :)